go.mod

```diff
@@ -3,80 +3,48 @@ module github.com/openshift/pagerduty-operator
 	github.com/go-logr/logr v0.1.0
-	github.com/go-openapi/jsonreference v0.19.3 // indirect
-	github.com/go-openapi/spec v0.19.5-0.20191022081736-744796356cda // indirect
-	github.com/gogo/protobuf v1.3.1 // indirect
-	github.com/golang/groupcache v0.0.0-20191027212112-611e8accdfc9 // indirect
 	github.com/golang/mock v1.3.1
-	github.com/google/uuid v1.1.1 // indirect
-	github.com/googleapis/gnostic v0.3.1 // indirect
-	github.com/gregjones/httpcache v0.0.0-20190611155906-901d90724c79 // indirect
-	github.com/hashicorp/golang-lru v0.5.3 // indirect
-	github.com/imdario/mergo v0.3.8 // indirect
-	github.com/json-iterator/go v1.1.8 // indirect
-	github.com/mailru/easyjson v0.7.0 // indirect
-	github.com/onsi/ginkgo v1.12.1 // indirect
-	github.com/onsi/gomega v1.10.0 // indirect
-	github.com/openshift/api v3.9.1-0.20190927182313-d4a64ec2cbd8+incompatible
+	github.com/openshift/api v3.9.1-0.20190924102528-32369d4db2ad+incompatible
 	github.com/openshift/cluster-network-operator v0.0.0-20190207145423-c226dcab667e // indirect
 	github.com/openshift/hive v0.0.0-20200128013012-99c7cccc7693
-	github.com/openshift/operator-custom-metrics v0.2.1
-	github.com/operator-framework/operator-sdk v0.8.2
-	github.com/pborman/uuid v0.0.0-20180906182336-adf5a7427709 // indirect
-	github.com/peterbourgon/diskv v2.0.1+incompatible // indirect
-	github.com/prometheus/client_golang v1.0.0
-	github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4 // indirect
-	github.com/prometheus/common v0.7.0 // indirect
-	github.com/prometheus/procfs v0.0.5 // indirect
+	github.com/openshift/operator-custom-metrics v0.3.0
+	github.com/operator-framework/operator-sdk v0.16.0
+	github.com/prometheus/client_golang v1.2.1
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.4.0
-	go.uber.org/multierr v1.2.0 // indirect
-	go.uber.org/zap v1.11.0 // indirect
-	golang.org/x/crypto v0.0.0-20191011191535-87dc89f01550 // indirect
-	golang.org/x/net v0.0.0-20191028085509-fe3aa8a45271 // indirect
-	golang.org/x/time v0.0.0-20191024005414-555d28b269f0 // indirect
-	google.golang.org/appengine v1.6.5 // indirect
-	gopkg.in/inf.v0 v0.9.1 // indirect
 	gotest.tools v2.2.0+incompatible
-	k8s.io/api v0.0.0-20190313235455-40a48860b5ab
-	k8s.io/apiextensions-apiserver v0.0.0-20190228180357-d002e88f6236 // indirect
-	k8s.io/apimachinery v0.0.0-20190313205120-d7deff9243b1
-	k8s.io/client-go v0.0.0-20181213151034-8d9ed539ba31
-	k8s.io/klog v1.0.0 // indirect
-	k8s.io/kube-openapi v0.0.0-20180711000925-0cf8f7e6ed1d // indirect
-	sigs.k8s.io/controller-runtime v0.1.12
-	sigs.k8s.io/yaml v1.1.0 // indirect
+	k8s.io/api v0.0.0
+	k8s.io/apimachinery v0.0.0
+	k8s.io/client-go v12.0.0+incompatible
+	sigs.k8s.io/controller-runtime v0.4.0
 )
 
-// Replaces from https://sdk.operatorframework.io/docs/migration/version-upgrade-guide/#v08x
-// Pinned to kubernetes-1.13.1
+// Replaces from https://sdk.operatorframework.io/docs/migration/version-upgrade-guide/#v016x
```

Comment:
Originally I didn't want to upgrade the operator-sdk at all (primarily because my understanding is that this operator still runs on OpenShift 3.11, and there's a bug in that version of k8s that will reject CRDs with some of the structured schema stuff that's generated by the newer versions of the operator-sdk), but I did it to pull in a newer version of the fake k8s client to be able to update some of the pre-existing tests to work with the new logic.

I could probably revert the operator-sdk upgrade & remove those tests as an alternative, would that be preferable?

Reply:
@jewzaam How do you feel about having the operator-sdk upgrade in here?
I'm not strongly opposed but would like extra eyes

Reply:
If we feel the upgrade will work on 3.11 I'm not opposed, it sounds like it's required for the testing added.  It could be pulled to a separate PR but I'd want it merged before we deploy these changes if it's adding useful tests.

Reply:
@jewzaam @RiRa12621 I've created #100 just now to give you the option to merge that separately if that's what you want to do. :) I won't make any change around it in this PR until you decide whether that's what you want to do or not, then I'll rebase this PR or not.